src/NumericalAlgorithms/Interpolation/Actions/InterpolationTargetVarsFromElement.hpp

```diff
@@ -0,0 +1,106 @@
+#include "NumericalAlgorithms/Interpolation/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace Parallel {
+template <typename Metavariables>
+class ConstGlobalCache;
+}  // namespace Parallel
+/// \endcond
+
+namespace intrp {
+namespace Actions {
+/// \ingroup ActionsGroup
+/// \brief Receives interpolated variables from an `Element` on a subset
+///  of the target points.
+///
+/// If interpolated variables for all target points have been received, then
+/// - Calls `InterpolationTargetTag::post_interpolation_callback`
+/// - Removes the finished `temporal_id` from `Tags::TemporalIds<TemporalId>`
+///   and adds it to `Tags::CompletedTemporalIds<TemporalId>`
+/// - Removes `Tags::InterpolatedVars<InterpolationTargetTag,TemporalId>`,
+///   `Tags::IndicesOfFilledInterpPoints`, and
+///   `Tags::IndicesOfInvalidInterpPoints` for the finished `temporal_id`.
+/// Uses:
+/// - DataBox:
+///   - `Tags::TemporalIds<TemporalId>`
+///   - `Tags::IndicesOfFilledInterpPoints<TemporalId>`
+///   - `Tags::InterpolatedVars<InterpolationTargetTag,TemporalId>`
+///   - `Tags::InterpolatedVars<InterpolationTargetTag,TemporalId>`
+///   - `::Tags::Variables<typename
+///                   InterpolationTargetTag::vars_to_interpolate_to_target>`
+///
+/// DataBox changes:
+/// - Adds: nothing
+/// - Removes: nothing
+/// - Modifies:
+///   - `Tags::TemporalIds<TemporalId>`
+///   - `Tags::CompletedTemporalIds<TemporalId>`
+///   - `Tags::IndicesOfFilledInterpPoints<TemporalId>`
+///   - `Tags::InterpolatedVars<InterpolationTargetTag,TemporalId>`
+///   - `::Tags::Variables<typename
+///                   InterpolationTargetTag::vars_to_interpolate_to_target>`
+///
+/// For requirements on InterpolationTargetTag, see InterpolationTarget
+template <typename InterpolationTargetTag>
+struct InterpolationTargetVarsFromElement {
+  /// For requirements on Metavariables, see InterpolationTarget
+  template <
+      typename ParallelComponent, typename DbTags, typename Metavariables,
+      typename ArrayIndex, typename TemporalId,
+      Requires<tmpl::list_contains_v<DbTags, Tags::TemporalIds<TemporalId>>> =
+          nullptr>
+  static void apply(
+      db::DataBox<DbTags>& box,
+      Parallel::ConstGlobalCache<Metavariables>& cache,
+      const ArrayIndex& /*array_index*/,
+      const std::vector<Variables<
+          typename InterpolationTargetTag::vars_to_interpolate_to_target>>&
+          vars_src,
+      const std::vector<std::vector<size_t>>& global_offsets,
+      const TemporalId& temporal_id) noexcept {
+    static_assert(
+        not InterpolationTargetTag::compute_target_points::is_sequential::value,
+        "Use InterpolationTargetGetVarsFromElement only with non-sequential"
+        " compute_target_points");
+
+    // Call set_up_interpolation only if it has not been called for this
+    // temporal_id.
+    if (not InterpolationTarget_detail::flag_temporal_ids_for_interpolation<
```

Comment:
please add a comment explaining why both clauses need to be true in order to execute the body (i.e. what happens if either or neither are true)

Reply:
Turns out that only the first clause is important.  If the first clause is true, then the second will always be true.  So I removed the second clause and added a comment.